src/iwant_bot/start.py

```diff
@@ -24,64 +51,336 @@ def format_message(message):
+    activities = '|'.join(_iwant_activities)    # Variable is used in pattern.
+    text_time = re.sub(f'({activities}|@[a-z0-9][-_.a-z0-9]{{1,20}}|with|or|and|[.!?]\s*$)', '', text)
+    print(text_time)
+    now_plus_duration = parse(text_time)  #zlobi... odmazat .,; a jine symboly, napsat testy.
+    print(now_plus_duration)
+    if now_plus_duration is None:
+        return _duration  # Implicit value
     else:
-        message = format_response(f'{user_name} wrote {text}')
+        # minimal valid time is 1s and maximal is _duration_max seconds
+        return min(max((now_plus_duration - datetime.now()).total_seconds(), 1.0), _duration_max)
+
+
+def iwant_invite(body: dict):
+    """Send notification to mentioned people, that someone wants to do something collectively."""
+    slack = Slack(BOT_TOKEN)
+    user_names = get_users(slack)
+    for name in user_names:
+        if re.search(f'@*{name[0]}', body['text']):
+            (text, attachment) = iwant_create_invitation(body, name[0], name[1])
+            slack.chat.post_message(channel=f'@{name[0]}',
+                                    text=text,
+                                    attachments=json.dumps(attachment))
+
+
+def iwant_create_invitation(body, name, name_id):
+    """Create message with buttons (accept or reject) for someone,
+    who was invited to join to some activity."""
+    time_minutes = round((body['duration_ts'] - body['incoming_ts']) / 60)
+    text = (f"Hi, @{body['user_name']} invites you to join "
+            f"{body['activity'][-1]} in {time_minutes} minutes.")
+    attachment = [
+        {
+            'text': 'Will you join?',
+            'callback_id': f"{body['callback_id'][-1]}",
+            'fallback': 'You do not want to join.',
+            #'color' : '#3AA3E3',
+            'attachment_type': 'default',
+            'actions': [
+                {
+                    'name': 'Invite',
+                    'text': 'Accept',
+                    'type': 'button',
+                    'value': f"{name} {name_id} {body['activity'][-1]} in {time_minutes} minutes."
+                },
+                {
+                    'name': 'Invite',
+                    'text': 'Reject',
+                    'type': 'button',
+                    'value': '0'
+                }
+            ]
+        }
+    ]
+    return (text, attachment)
+
+
+# button = [
+#     {'text': 'Testing ok button',
+#      "callback_id": "test11", 'actions': [
+#             {'name': 'test', 'text': 'ok', 'type': 'button', 'value': 1},
+#             {'name': 'test', 'text': 'no', 'type': 'button', 'value': 0}
+#         ]
+#     },
+#     {'text': 'Test 2 "ok button"',
+#      "callback_id": "test22", 'actions': [
+#         {'name': 'test', 'text': 'OK', 'type': 'button', 'value': 1},
+#         {'name': 'test', 'text': 'NO', 'type': 'button', 'value': 0}
+#     ]
+#     }
+# ]
+# print(json.dumps(button))
+
+
+
+def get_users(slack):
```

Comment:
This function (and other as well) use global objects s.a. `slack`. Although it is probably OK now, it is a good idea to write a TODO that the architecture should be refactored to get rid of those.